state/factory/statedb.go

```diff
@@ -102,38 +101,38 @@ func CachedStateDBOption() StateDBOption {
 
 // InMemStateDBOption creates in memory state db
 func InMemStateDBOption() StateDBOption {
-	return func(sdb *stateDB, cfg config.Config) error {
+	return func(sdb *stateDB, cfg *Config) error {
 		sdb.dao = db.NewMemKVStore()
 		return nil
 	}
 }
 
 // RegistryStateDBOption sets the registry in state db
 func RegistryStateDBOption(reg *protocol.Registry) StateDBOption {
-	return func(sdb *stateDB, cfg config.Config) error {
+	return func(sdb *stateDB, cfg *Config) error {
 		sdb.registry = reg
 		return nil
 	}
 }
 
 // SkipBlockValidationStateDBOption skips block validation on PutBlock
 func SkipBlockValidationStateDBOption() StateDBOption {
-	return func(sdb *stateDB, cfg config.Config) error {
+	return func(sdb *stateDB, cfg *Config) error {
 		sdb.skipBlockValidationOnPut = true
 		return nil
 	}
 }
 
 // DisableWorkingSetCacheOption disable workingset cache
 func DisableWorkingSetCacheOption() StateDBOption {
-	return func(sdb *stateDB, cfg config.Config) error {
+	return func(sdb *stateDB, cfg *Config) error {
 		sdb.workingsets = cache.NewDummyLruCache()
 		return nil
 	}
 }
 
 // NewStateDB creates a new state db
-func NewStateDB(cfg config.Config, opts ...StateDBOption) (Factory, error) {
+func NewStateDB(cfg Config, opts ...StateDBOption) (Factory, error) {
```

Comment:
@dustinxie 
Could `NewStateDB` be changed to `func NewStateDB(cfg config.Config, dao db.KVStore, opts ...StateDBOption) (Factory, error)`?
Then all opts related to dao could be removed

Reply:
it's possible, yet that means moving these logic like below outside
```
cfg.DB.DbPath = cfg.Chain.TrieDBPath
```
and create the `KVStore` outside, which incurs lots of additional change.

so IMO that's not very necessary? thx for the observation though

Reply:
had another discussion, if we do this, we could remove `db.Config` from `factory.Config`, so I am fine to do this as well
maybe open another PR right after this, or add another commit, either way is fine